test/system/test_api_interfaces.py

```diff
@@ -91,10 +91,62 @@ def test_delete_and_return_true(self):
             self.assertNotIn('Loopback0', config[0]['interfaces'])
 
     def test_delete_ethernet_raises_not_implemented_error(self):
-        with self.assertRaises(NotImplementedError):
-            for dut in self.duts:
+        for dut in self.duts:
+            with self.assertRaises(NotImplementedError):
                 dut.api('interfaces').delete(random_interface(dut))
 
+    def test_create_and_delete_ethernet_sub_interface(self):
```

Comment:
What do you mean by this? Do you mean running create on a physical port like 'Ethernet1' or simply creating the subinterface before the primary interface is in routed mode? EOS allows the creation of the subinterface before the primary is in routed mode.